src/database/models/notification_item.rs

```diff
@@ -30,52 +36,64 @@ impl NotificationBuilder {
         &self,
         user: UserId,
         transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+        redis: &deadpool_redis::Pool,
     ) -> Result<(), DatabaseError> {
-        self.insert_many(vec![user], transaction).await
+        self.insert_many(vec![user], transaction, redis).await
     }
 
     pub async fn insert_many(
         &self,
         users: Vec<UserId>,
         transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+        redis: &deadpool_redis::Pool,
     ) -> Result<(), DatabaseError> {
+        let mut notifications = Vec::new();
         for user in users {
             let id = generate_notification_id(&mut *transaction).await?;
 
-            Notification {
+            notifications.push(Notification {
                 id,
                 user_id: user,
                 body: self.body.clone(),
                 read: false,
                 created: Utc::now(),
-            }
-            .insert(&mut *transaction)
-            .await?;
+            });
         }
 
+        Notification::insert_many(&notifications, transaction, redis).await?;
+
         Ok(())
     }
 }
 
 impl Notification {
-    pub async fn insert(
-        &self,
+    pub async fn insert_many(
```

Comment:
I did this for a couple reasons:
- The only caller of `insert` was `NotificationBuilder::insert_many`
- We should be batching our Postgres inserts using this: https://github.com/launchbadge/sqlx/blob/main/FAQ.md#how-can-i-bind-an-array-to-a-values-clause-how-can-i-do-bulk-inserts
- This allows me to batch the delete from redis while maintaining only a single code path. If the single insert had been allowed to continue, I could have done the cache clearing up at `NotificationBuilder::insert_many`, but then that would have risked `Notification::insert` being called and not clearing the cache

Reply:
Yup this makes sense to me. I would also agree reformatting bulk inserts should be its own PR

Reply:
Per discussion with Jai, I went ahead and did it. Completely untested, intending to test with #719